packages/cursorless-engine/src/actions/GenerateSnippet/GenerateSnippet.ts

```diff
@@ -1,263 +1,25 @@
-
-    /**
-     * Constructs a key-value entry for use in the variable description section
-     * of the user snippet definition.  It contains tabstops for use in the
-     * meta-snippet.
-     * @param variable The variable
-     * @returns A [key, value] pair for use in the meta-snippet
-     */
-    const constructVariableDescriptionEntry = ({
-      placeholderIndex,
-    }: Variable): [string, string] => {
-      // The key will have the same placeholder index as the other location
-      // where this variable appears.
-      const key = "$" + placeholderIndex;
-
-      // The value will end up being an empty object with a tabstop in the
-      // middle so that the user can add information about the variable, such
-      // as wrapperScopeType.  Ie the output will look like `{|}` (with the `|`
-      // representing a tabstop in the meta-snippet)
-      //
-      // NB: We use the substituter here, with `isQuoted=true` because in order
-      // to make this work for the meta-snippet, we want to end up with
-      // something like `{$3}`, which is not valid json.  So we instead arrange
-      // to end up with json like `"hgidfsivhs"`, and then replace the whole
-      // string (including quotes) with `{$3}` after json-ification
-      const value = substituter.addSubstitution(
-        "{$" + currentPlaceholderIndex++ + "}",
-        true,
-      );
-
-      return [key, value];
-    };
-
-    /** An object that will be json-ified to become the meta-snippet */
-    const snippet = {
-      [snippetName]: {
-        definitions: [
-          {
-            scope: {
-              langIds: [editor.document.languageId],
-            },
-            body: snippetLines,
-          },
-        ],
-        description: "$" + currentPlaceholderIndex++,
-        variables:
-          variables.length === 0
-            ? undefined
-            : Object.fromEntries(
-                variables.map(constructVariableDescriptionEntry),
-              ),
-      },
-    };
-
-    /**
-     * This is the text of the meta-snippet in Textmate format that we will
-     * insert into the new document where the user will fill out their snippet
-     * definition
-     */
-    const snippetText = substituter.makeSubstitutions(
-      JSON.stringify(snippet, null, 2),
-    );
-
-    const editableEditor = ide().getEditableTextEditor(editor);
-
-    if (ide().runMode === "test") {
-      // If we're testing, we just overwrite the current document
-      await editableEditor.setSelections([
-        editor.document.range.toSelection(false),
-      ]);
-    } else {
-      // Otherwise, we create and open a new document for the snippet in the
-      // user snippets dir
-      await this.snippets.openNewSnippetFile(snippetName);
+    if (dirPath == null) {
+      const action = new GenerateSnippetLegacy(this.snippets);
+      return action.run(targets, snippetName);
     }
 
-    // Insert the meta-snippet
-    await editableEditor.insertSnippet(snippetText);
```

Comment:
NB: this was technically a bug and relying on odd Visual Studio Code behavior because we aren't using the new editor that was created inside of `this.snippets.openNewSnippetFile(snippetName);`. It just so happens that `insertSnippet` always uses the current editor even if you call it on a different one. The new code fixes this by making the function return a promise with the actual new editor which we needed for appending.

Reply:
NB: Andreas fixed this because we now want to append the text to the file if it already exists, and reading the text from an editor that's not the current one does work, and that's when he found that the editor object was actually technically wrong